### ISSUE TITLE
mips32r6 build fails Octane Box2D in 4.4 branch

### ISSUE THREAD

**Issue by paul99** [v8mips/v8mips#142]
I pulled 4.4 branch, and first off it is broken with gcc 4.9.2 for -Wsign-compare issues. The change is the part of Wingo's fix `004bb77e Fix more -Werror=sign-compare bugs with GCC 4.9.2`, but just for the file counters-unittest.cc. I dropped a patch to Dropbox mips folder: v8-4.4-sign-cmp-fix.diff (we need to figure out how to back-merge this 'part of a master-branch patch' back to 4.4.)

I then built for mips32r6 and qemu user-mode -- and when running Octane, it gets stuck in infinite loop in Box2D. If I comment out that test, it will finish the rest of Octane. 

Mips64r6 works fine. Branches 4.1, 4.2, 4.3 work fine. I did not test master, but suspect problem may well exist there too.

**Comment by dusmil** [comment]
Problem reproduced on simulator for branch 4.4.9. The octane benchmark never ends after Box2d is started. The mips32r6 build from current master works fine on simulator.

it appears it is a missing commit, but it does not make sense if misp64r6 works.

**Comment by ilijapavlovic** [comment]
The last branch where the problem was possible to reproduce is 4.4.31.
In branch 4.4.32 benchmark tests are finished without problems and this branch contains following commit:

```
commit aaddea11b225fd9b004f5cb6e572c1d74c23fcfb
Author: jarin <jarin@chromium.org>
Date:   Thu Apr 23 01:07:12 2015 -0700

    Materialize booleans in the turbofan deoptimizer.

    BUG=
    R=titzer@chromium.org

    Review URL: https://codereview.chromium.org/1055453006

    Cr-Commit-Position: refs/heads/master@{#28022}

```

**Comment by ilijapavlovic** [comment]
During attempts to reproduce infinite loops in benchmark tests, it is noted that test Box2D in most cases was not in the loop. Very often it happened at test Typescript and rarely on other tests (PdfJS, DeltaBlue, Splay).

**Comment by paul99** [comment]
@ilijapavlovic does this need more testing or followup? Or is it fixed?

**Comment by ilijapavlovic** [comment]
No. It does not need more testing. This problem is fixed in the branch 4.4.32.
This issues can be closed. The main goal of my work here was to find which commit solves this problem.

**Comment by paul99** [comment]
Closing.
